src/Sarif.Converters.UnitTests/FortifyFprConverterTests.cs

```diff
@@ -1,13 +1,16 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using Microsoft.CodeAnalysis.Sarif.TestUtilities;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Microsoft.CodeAnalysis.Sarif.Converters
 {
-    public class FortifyFprConverterTests : FileDiffingTests
+    public class FortifyFprConverterTests : FileDiffingTests, IClassFixture<FortifyFprConverterTests.FortifyFprConverterTestsFixture>
     {
+        public class FortifyFprConverterTestsFixture : DeletesOutputsDirectoryOnClassInitializationFixture { }
```

Comment:
>DeletesOutputsDirectoryOnClassInitializationFixture [](start = 55, length = 51)

This fixture deletes the outputs directory once per test session. Previously, we deleted this directory on executing every test case. Oops. This means that only the very last test failure was persisted to the directory.

Reply:
Oh, nice, I will add that to SarifCurrentToVersionOneVisitorTests.

---
In reply to: [247381989](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247381989) [](ancestors = 247381989)

Reply:
Just want to be clear you understand this pattern. A fixture is called on a class with a public parameterless constructor. it is instantiated a single time on xunit identifying a class with one or more [Fact] or other attribute-derived tests. The test class itself cannot be the fixture class (as xUnit supports only one public ctor on a test class, due to the diversity of parameterizations of these things, I guess. Our fixture derives some computed data (i.e., the test output directory) from our test organization scheme. From the base class. I have recapped this minimal data computation and chosen to embed these fixtures as nested types with the actual test class (since they are not generally repurposable. The fixture must be named identically to the test class with the term 'Fixture' appended to it.

If you can think of anything cleaner/more elegant, let me know. The fixture itself can be provided, btw, to the test class ctor, with the result that it can be additionally parameterized when it is passed to each test case. This parameterization can be used most easily for test cleanup. You could imagine us simply placing a bool property on this thing, though ('HaveIDeletedTheOutputDirectoryYet' and having the first test case note that it's false, perform the delete itself, and set that bool to true. The test fixture is effectively just a bundle of static data with this approach. It would eliminate the requirement to define a nested type for each test (because we could put this standard operation into the file diffing base type). Thoughts?


---
In reply to: [247657934](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247657934) [](ancestors = 247657934,247381989)

Reply:
I'll read up on XUnit's "fixture" support and get back to you.

---
In reply to: [247661342](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247661342) [](ancestors = 247661342,247657934,247381989)

Reply:
I read the XUnit documentation of [shared context](https://xunit.github.io/docs/shared-context). Now I know enough to ask:

Why do you derive a class from `DeletesOutputsDirectoryOnClassInitializationFixture`? I could understand it if the derived class added some functionality, but it doesn't. Why not just:

```
FortifyFprConverterTests : FileDiffingTests, IClassFixture<DeletesOutputsDirectoryOnClassInitializationFixture>
{
    ...
}
```

---
In reply to: [247662456](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247662456) [](ancestors = 247662456,247661342,247657934,247381989)

Reply:
Ok, I'm bouncing back and forth between commenting on the fixture and commenting on this class :-). I now see that the fixture has virtual properties that a `FileDiffingTests`-derived class can override in a bespoke derived fixture class. But in cases like this class, that do _not_ override those properties or do anything else, I still think you should just mark them with `IClassFixture<DeletesOutputsDirectoryOnClassInitializationFixture>`.

---
In reply to: [247724801](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247724801) [](ancestors = 247724801,247662456,247661342,247657934,247381989)

Reply:
No, this won't work. These properties are initialized based on class naming convention. A single class can only represent the intended output location of a single suite. I'm just going to change the pattern entirely, stay tuned. Not this change.

---
In reply to: [247732570](https://github.com/Microsoft/sarif-sdk/pull/1212#discussion_r247732570) [](ancestors = 247732570,247724801,247662456,247661342,247657934,247381989)